R-package/src/lightgbm_R.cpp

```diff
@@ -1103,6 +1257,32 @@ SEXP LGBM_BoosterSaveModel_R(SEXP handle,
   R_API_END();
 }
 
+// Note: for some reason, MSVC crashes when an error is thrown here
+// if the buffer variable is defined as 'std::unique_ptr<std::vector<char>>',
+// but not if it is defined as '<std::vector<char>'.
+#ifndef _MSC_VER
+SEXP LGBM_BoosterSaveModelToString_R(SEXP handle,
+  SEXP num_iteration,
+  SEXP feature_importance_type) {
```

Comment:
I've just merged #6398. In that, @mayer79 exposed `start_iteration` on `LGBM_BoosterSaveModelToString_R()`. Could you add that on this implementation?

Sorry for that, it's unfortunate timing that these two PRs both modifying `LGBM_BoosterSaveModelToString_R()` reached the ready-to-be-merged state at roughly the same time.